docs/Installation.rst

```diff
@@ -178,25 +121,142 @@ The following command should now mount the cgroup filesystem:
+
+The quick and dirty way of installing Python dependencies is this:
+
+.. sourcecode:: bash
+
+    pip install --user -r requirements.txt
+
+This command installs python dependencies in your home folder. If you really want to install them globally then you should remove ``--user`` and run the install command as root (but that could be a bad idea depending on your Linux distribution, because you would be installing files in a system directory bypassing the system package manager).
+
+There are other ways to manage python dependencies:
+
+virtualenv
+----------
+
+A `virtual environment <https://virtualenv.pypa.io/en/latest/>`_ is an isolated Python environment that you can put wherever you like and that can be "activated" and "deactivated" at will. The tool you need in order to create a virtual environment is called ``virtualenv``, and can be installed by looking for ``python-virtualenv`` using your Linux distribution's package manager. For example:
+
+* Ubuntu: `python-virtualenv <https://apps.ubuntu.com/cat/applications/python-virtualenv/>`_.
+* Arch Linux: `python-virtualenv <https://www.archlinux.org/packages/extra/any/python-virtualenv/>`_.
+
+.. FUTURE FIXME: virtualenv installation is necessary only on python2; when the
+   porting to python3 will be complete, the "new" way of creating a virtual
+   environment will be ``pyvenv`` or equivalently ``python -m venv`` (the venv
+   module and the pyvenv script come bundled with python3, so there's no need to
+   install virtualenv anymore).
+
+Once you installed ``virtualenv``, you will need to create a virtual environment somewhere in your filesystem. For example, let's assume that you decided to create it under your home directory (as ``~/cms_venv``):
+
+.. sourcecode:: bash
+
+    virtualenv -p python2 ~/cms_venv
+
+To "activate" it:
 
-If you prefer using Python Package Index, you can retrieve all Python dependencies with this line:
+.. sourcecode:: bash
+
+    . ~/cms_venv/bin/activate
+
+After the activation, python commands (python, pip) will refer to their corresponding virtual version, so you can install python dependencies by issuing:
+
+.. sourcecode:: bash
+
+    pip install -r requirements.txt
+
+.. note::
+
+    Once you finished installing CMS (and using it) you can deactivate the virtual environment by issuing:
+
+    .. sourcecode:: bash
+
+        deactivate
+
+Ubuntu
+------
+
+.. warning::
+
+  It's usually possible to install python dependencies using your Linux distribution's package manager. However, keep in mind that the version of each package is controlled by the package mantainers and could be too new or too old for CMS. This is generally not the case on Ubuntu (since the package versions are usually stable and quite old).
```

Comment:
This is not generally the case on Ubuntu because we try to build on the packages available at a certain LTS version (in this case is 14.04), not because the packages are old. Please rephrase.